pyanaconda/format_dasd.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Vendula Poncova <vponcova@redhat.com>
+#
+
+import gi
+gi.require_version("BlockDev", "2.0")
+from gi.repository import BlockDev as blockdev
+
+from blivet import arch
+from blivet.osinstall import storage_initialize
+
+from pyanaconda.flags import flags
+from pyanaconda.ui.lib.disks import getDisks
+from pyanaconda.isignal import Signal
+from pyanaconda.storage_utils import on_disk_storage
+from pyanaconda.i18n import _
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class DasdFormatting(object):
+    """Class for formatting DASDs."""
+
+    def __init__(self):
+        self._dasds = []
+
+        self._can_format_unformatted = True
+        self._can_format_ldl = True
+
+        self._report = Signal()
+        self._report.connect(log.debug)
+
+    @staticmethod
+    def is_supported():
+        """Is DASD formatting supported on this machine?"""
+        return arch.is_s390()
+
+    @property
+    def report(self):
+        """Signal for the progress reporting.
+
+        Emits messages during the formatting.
+        """
+        return self._report
+
+    @property
+    def dasds(self):
+        """List of found DASDs to format."""
+        return self._dasds
+
+    @property
+    def dasds_summary(self):
+        """Returns a string summary of DASDs to format."""
+        return "\n".join(map(self.get_dasd_info, self.dasds))
+
+    def get_dasd_info(self, disk):
+        """Returns a string with description of a DASD,"""
```

Comment:
I guess this should be `.` instead of `,`.

Reply:
Fixed.